### PR TITLE
Configurable AP passphrase

### PR DESCRIPTION
**Pull request by ceesb**
I added a configurable passphrase for WiFi AP under the "advanced settings" menu.

### CODE REVIEW

**Comment by glynhudson** [comment]
I notice you have included a change to point the GUI sub-module to https://github.com/ceesb/openevse_wifi_gui, I don't think this was intentional. Can you amend?

**Comment by ceesb** [comment]
Ah sorry, I fixed that. The sub-module stuff is confusing when you fork it ;)

There is another issue that I'm unable to fix, and that is the (client side) validation of input fields in the web interface. This is especially relevant for the AP passphrase, since you need more than 8 characters or the AP will not come up anymore. I looked into "knockout-validation" to get the input fields validated client side, but I lack the web development skills to enable this correctly. 

If you want, I can add another save-ap-passhphrase end-point in the webserver that does validation server side, but than I can't have the passphrase field under the "advanced settings" tab.

**Comment by glynhudson** [comment]
What do you think of this PR @jeremypoulter? This would be a good feature to add. Would be best to wait until V4.0 is ready?

File: src/app_config.cpp [hunk 1]
```diff
@@ -77,6 +82,7 @@ ConfigOpt *opts[] =
 // Wifi Network Strings
   new ConfigOptDefenition<String>(esid, "", "ssid", "ws"),
   new ConfigOptSecret(epass, "", "pass", "wp"),
+  new ConfigOptDefenition<String>(apass, "openevse", "apass", "apass"),
```
Comment (jeremypoulter) [inline src/app_config.cpp:85]: The second name should be short to save flash space but the first can be a bit more descriptive
```suggestion
  new ConfigOptDefenition<String>(apass, "openevse", "ap_pass", "ps"),
```

**Comment by jeremypoulter** [comment]
Can add to v3 or v4 does not look to have anything that would conflict. just a minor change suggested

**Comment by jeremypoulter** [comment]
Oh and the issue @ceesb mentions about validation, I will give it some consideration on how to add validation to the generic endpoint, probably more than just the AP password should be validated. In terms of doing it in the GUI, there is an example here: https://github.com/openevse/openevse_wifi_gui/blob/wifi_timer_updates/src/view_models/OpenEvseWiFiViewModel.js#L466-L477

**Comment by jeremypoulter** [comment]
Superceeded by #595

**Pull request closed without merging**